src/main/java/uk/gov/ons/census/action/messaging/FulfilmentRequestReceiver.java

```diff
@@ -0,0 +1,169 @@
+        .debug("Fulfilment Requested Event");
+    if (fulfilmentCase.isEmpty()) {
+      log.with("caseId", caseId).error("Cannot find Case for fulfilment request.");
+      throw new RuntimeException(
+          String.format("Cannot find case %s for fulfilment request.", caseId));
+    }
+    return fulfilmentCase.get();
+  }
+
+  private PrintFileDto createAndPopulatePrintFileDto(
+      Case fulfilmentCase, ActionType actionType, ResponseManagementEvent event) {
+    PrintFileDto printFileDto = new PrintFileDto();
+    printFileDto.setAddressLine1(fulfilmentCase.getAddressLine1());
+    printFileDto.setAddressLine2(fulfilmentCase.getAddressLine2());
+    printFileDto.setAddressLine3(fulfilmentCase.getAddressLine3());
+    printFileDto.setTownName(fulfilmentCase.getTownName());
+    printFileDto.setPostcode(fulfilmentCase.getPostcode());
+    printFileDto.setTitle(event.getPayload().getFulfilmentRequest().getContact().getTitle());
+    printFileDto.setForename(event.getPayload().getFulfilmentRequest().getContact().getForename());
+    printFileDto.setSurname(event.getPayload().getFulfilmentRequest().getContact().getSurname());
+    printFileDto.setBatchId(UUID.randomUUID().toString());
+    printFileDto.setBatchQuantity(1);
+    printFileDto.setPackCode(event.getPayload().getFulfilmentRequest().getFulfilmentCode());
+    printFileDto.setActionType(actionType.name());
+    printFileDto.setCaseRef(fulfilmentCase.getCaseRef());
+    return printFileDto;
+  }
+
+  private Optional<ActionType> determineActionType(String fulfilmentCode)
+      throws UnknownFulfilmentException {
+    switch (fulfilmentCode) {
+      case "P_OR_H1":
+      case "P_OR_H2":
+      case "P_OR_H2W":
+      case "P_OR_H4":
+        return Optional.of(ActionType.P_OR_HX);
+      case "P_LP_HL1":
+      case "P_LP_HL2":
+      case "P_LP_HL2W":
+      case "P_LP_HL4":
+        return Optional.of(ActionType.P_LP_HLX);
+      case "P_TB_TBARA1":
+      case "P_TB_TBBEN1":
+      case "P_TB_TBCAN1":
+      case "P_TB_TBCAN4":
+      case "P_TB_TBFRE1":
+      case "P_TB_TBGUJ1":
+      case "P_TB_TBGUR1":
+      case "P_TB_TBIRI4":
+      case "P_TB_TBITA1":
+      case "P_TB_TBKUR1":
+      case "P_TB_TBLIT1":
+      case "T_PB_TBLIT4":
+      case "P_TB_TBMAN1":
+      case "P_TB_TBMAN4":
+      case "P_TB_TBPOL1":
+      case "P_TB_TBPOL4":
+      case "P_TB_TBPOR1":
+      case "P_TB_TBRUS1":
+      case "P_TB_TBURD1":
+      case "P_TB_TBSOM1":
+      case "P_TB_TBSPA1":
+      case "P_TB_TBTUR1":
+      case "P_TB_TBULS4":
+      case "P_TB_TBVIE1":
+      case "P_TB_TBYSH1":
+        return Optional.of(ActionType.P_TB_TBX);
+      case "UACHHT1":
+      case "UACHHT2":
+      case "UACHHT2W":
+      case "UACHHT4":
+      case "UACIT1":
+      case "UACIT2":
+      case "UACIT2W":
+      case "UACIT4":
+        return Optional.empty(); // Ignore SMS fulfilments
+      default:
+        throw new UnknownFulfilmentException(fulfilmentCode);
```

Comment:
nope. If a runtime exception is what we want, throw one here. If we want to ignore it return empty. We're going down the road of mountains of YAGNI code.

Reply:
if you want to log stuff, here would be a good place to do it.